src/map/vending.cpp

```diff
@@ -344,17 +351,37 @@ int8 vending_openvending( map_session_data& sd, const char* message, const uint8
 		sd.vending[i].index = index;
 		sd.vending[i].amount = amount;
 		sd.vending[i].value = min(value, (unsigned int)battle_config.vending_max_value);
+		total += static_cast<int64>(sd.vending[i].value) * amount;
 		i++; // item successfully added
 	}
 
+	// check if the total value of the items plus the current zeny is over the limit
+	if ( !battle_config.vending_over_max && (static_cast<int64>(sd.status.zeny) + total) > MAX_ZENY ) {
+#if PACKETVER >= 20200819
+		clif_msg_color( &sd, MSI_MERCHANTSHOP_TOTA_LOVER_ZENY_ERR, color_table[COLOR_RED] );
+#else
+		clif_skill_fail( sd, MC_VENDING );
+#endif
```

Comment:
```suggestion
#endif
		clif_skill_fail( sd, MC_VENDING );
```
I would keep on sending this, just like in all other cases.